app/controllers/miq_ae_tools_controller.rb

```diff
@@ -127,9 +127,13 @@ def cancel_import
   end
 
   def import_via_git
-    git_based_domain_import_service.import(params[:git_repo_id], params[:git_branch_or_tag], current_tenant.id)
+    begin
+      git_based_domain_import_service.import(params[:git_repo_id], params[:git_branch_or_tag], current_tenant.id)
 
-    add_flash(_("Imported from git"), :info)
+      add_flash(_("Imported from git"), :info)
+    rescue MiqException::Error
+      add_flash(_("The selected branch or tag does not have valid domain object data"), :error)
```

Comment:
@eclarizio 
we could fix the error message that is being generated since we can all kinds of errors, e.g. permissions, corrupted repository etc. So it would make sense to leave a generic error message on the UI side and fix the code that raises the exception to have a more meaningful message